test/zkbob/utils/ZkBobAccounting.t.sol

```diff
@@ -504,6 +508,225 @@ contract ZkBobAccountingTest is Test {
         assertEq(limits3.tier, 255);
     }
 
+    function _setKYCPorviderManager() internal returns (SimpleKYCProviderManager) {
+        ERC721PresetMinterPauserAutoId nft = new ERC721PresetMinterPauserAutoId("Test NFT", "tNFT", "http://nft.url/");
+
+        SimpleKYCProviderManager manager = new SimpleKYCProviderManager(nft, TIER_FOR_KYC);
+        pool.setKycProvidersManager(manager);
+
+        return manager;
+    }
+
+    function _mintNFT(ERC721PresetMinterPauserAutoId _nft, address _user) internal returns (uint256) {
+        uint256 tokenId = 0;
+        vm.recordLogs();
+        _nft.mint(_user);
+        Vm.Log[] memory entries = vm.getRecordedLogs();
+        for (uint256 i = 0; i < entries.length; i++) {
+            if (entries[i].topics[0] == keccak256("Transfer(address,address,uint256)")) {
+                tokenId = uint256(entries[i].topics[3]);
+                emit log_uint(tokenId);
+                break;
+            }
+            // fail test if the event is not found
+            assertLt(i, entries.length - 1);
+        }
+        return tokenId;
+    }
+
+    function testSetKycProvidersManager() public {
+        address manager = address(_setKYCPorviderManager());
+        assertEq(address(pool.kycProvidersManager()), manager);
+
+        vm.expectRevert("ZkBobPool: manager is zero address");
+        pool.setKycProvidersManager(SimpleKYCProviderManager(address(0)));
+    }
+
+    function testGetLimitsForTiersWithKYCProvider() public {
+        ZkBobAccounting.Limits memory limits;
+
+        SimpleKYCProviderManager manager = _setKYCPorviderManager();
+        ERC721PresetMinterPauserAutoId nft = ERC721PresetMinterPauserAutoId(address(manager.NFT()));
+
+        //                            dailyDepositCap                  depositCap
+        //                            |          dailyWithdrawalCap    |          dailyUserDirectDepositCap
+        //                tvlCap      |          |          dailyUserDepositCap   |  directDepositCap
+        pool.setLimits(0, 1000 ether, 500 ether, 400 ether, 300 ether, 150 ether, 0, 0);
```

Comment:
https://docs.soliditylang.org/en/v0.8.18/control-structures.html#function-calls-with-named-parameters

Reply:
Interesting. Let's try to improve the code reading in this way: https://github.com/zkBob/zkbob-contracts/pull/54/commits/5639b1f0dd92683957d2a6bfaac7bd408f403027